sunpy/net/dataretriever/sources/gong_synoptic.py

```diff
@@ -0,0 +1,46 @@
+from sunpy.net.dataretriever import GenericClient
+
+__all__ = ['GongSynopticClient']
+
+
+class GongSynopticClient(GenericClient):
+    """
+    Provides access to the Magnetogram products of NSO-GONG synoptic Maps.
+
+    Searches data hosted by the `National Solar Observatory <gong2.nso.edu/oQR/zqs/>`__
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido, attrs as a
+    >>> results = Fido.search(a.Time("2019/12/31 22:00", "2020/1/1 02:00"),
+    ...                       a.Instrument('GONG'))  #doctest: +REMOTE_DATA
+    >>> results  #doctest: +REMOTE_DATA
+    <sunpy.net.fido_factory.UnifiedResponse object at ...>
+    Results from 1 Provider:
+    <BLANKLINE>
+    3 Results from the GongSynopticClient:
+         Start Time           End Time      Instrument ... Source ExtentType
+    ------------------- ------------------- ---------- ... ------ ----------
+    2019-12-31 22:14:00 2019-12-31 22:14:59       GONG ...    NSO   SYNOPTIC
+    2019-12-31 23:04:00 2019-12-31 23:04:59       GONG ...    NSO   SYNOPTIC
+    2019-12-31 23:54:00 2019-12-31 23:54:59       GONG ...    NSO   SYNOPTIC
+    <BLANKLINE>
+    <BLANKLINE>
+
+    """
+    baseurl = (r'https://gong2.nso.edu/oQR/zqs/%Y%m/mrzqs%y%m%d/mrzqs%y%m%dt%H%Mc'
+               r'(\d){4}_(\d){3}\.fits.gz')
+    pattern = '{}/zqs/{year:4d}{month:2d}/mrzqs{:4d}{day:2d}/mrzqs{:6d}t{hour:2d}{minute:2d}c{}'
+
+    @classmethod
+    def _attrs_module(cls):
+        return 'gong_synoptic', 'sunpy.net.dataretriever.attrs.gong_synoptic'
+
+    @classmethod
+    def register_values(cls):
+        from sunpy.net import attrs
+        adict = {attrs.Instrument: [("GONG", "Global Oscillation Network Group.")],
+                 attrs.Physobs: [("LOS_MAGNETIC_FIELD", "Line of sight magnetic field")],
+                 attrs.Source: [('NSO', 'National Solar Observatory.')],
+                 attrs.gong_synoptic.ExtentType: [("SYNOPTIC", "Coverage of a complete solar rotation synthesized over time")]}
```

Comment:
I don't have a better suggestion but `attrs.gong_synoptic.ExtentType` seems a bit long.

Reply:
Can we just remove it?
It will only happen that there won't be an extra column of name `extent type` in results whoes value is "synoptic". 
Nothing else would change.

Reply:
I think it would be reasonable to move `ExtentType` out of `attrs.gong_synoptic` and into `attrs` as a more generic attribute, don't know what others think about that?

Reply:
IF we need to use it for this client, I would be in favour of that instead of it being for this client only.

Reply:
Given one can also get full disc magnetograms from GONG, I actually think we should rename the client from `gong_synoptic` > `gong`, and keep this keyword as a way to differentiate between a full disk map and a synoptic map. Then we avoid having separate `gong` and a `gong_synoptic` clients.

Reply:
Replaced gongsynoptic with gong and moved extentType to attrs.